### PR TITLE
Chore: Replace go-multierror with errors package

### LINKED ISSUE #63686: Replace go-multierror with errors package in Go 1.20
We should replace usage of `go-multierror` with the errors package once Grafana moves to Go 1.20. For example, using https://pkg.go.dev/errors#Join and `fmt.Errorf` with more than one `%w`.

### PR DESCRIPTION
**Pull request by ghost**
**What is this feature?**

Replaced go-multierror package with errors package.

**Why do we need this feature?**

**Who is this feature for?**

**Which issue(s) does this PR fix?**:

Fixes #63686

**Special notes for your reviewer:**

Please check that:
- [x] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

**Review by papagian — COMMENTED** [review]
Thank you for your contribution! I think that, now, after [upgrading the target Go version](https://github.com/grafana/grafana/pull/67102) to 1.20 we can merge it.
it would be nice to update also the go.mod (by syncing your branch with the latest `main` and run `go mod tidy`) so that the dependency becomes indirect.

**Comment by papagian** [comment]
@iSatVeerSingh have you closed it by mistake? I think that we a few additional changes we could have merged it
shall I re-open it?

**Comment by papagian** [comment]
@iSatVeerSingh can you try to follow the instructions and resolve the conflict: the `pkg/codegen/diffwrite.go` has been deleted from the main branch therefore you need to remove it also from your branch
(I don't have permission to push to your branch to update it myself)

also can you please run `go mod tidy` and push the change in the go.mod file as well, thank you!

**Comment by iSatVeerSingh** [comment]
@papagian I run go mod tidy. It is adding //indirect instead of removing package

**Comment by grobinson-grafana** [comment]
Hi! :wave: You have some lint errors that are causing the build to fail:

```
pkg/services/ngalert/state/cache_test.go:69:3: S1021: should merge variable declaration with assignment on next line (gosimple)
		var multierr []error
		^
pkg/services/ngalert/state/cache_test.go:105:3: S1021: should merge variable declaration with assignment on next line (gosimple)
		var multierr []error
		^
pkg/services/ngalert/state/cache_test.go:71:19: errorsas: second argument to errors.As must be a non-nil pointer to either a type that implements error, or to any interface type (govet)
		require.True(t, errors.As(err, &multierr))
		                ^
pkg/services/ngalert/state/cache_test.go:107:19: errorsas: second argument to errors.As must be a non-nil pointer to either a type that implements error, or to any interface type (govet)
		require.True(t, errors.As(err, &multierr))
```

**Comment by papagian** [comment]
> @papagian I run go mod tidy. It is adding //indirect instead of removing package

I think, that's fine; it's because another library we import uses it:
```
go mod why github.com/hashicorp/go-multierror
# github.com/hashicorp/go-multierror
github.com/grafana/grafana/pkg/codegen
github.com/grafana/codejen
github.com/hashicorp/go-multierror
```

File: pkg/services/ngalert/state/cache_test.go [hunk 1]
```diff
@@ -67,14 +66,14 @@ func Test_expand(t *testing.T) {
 		require.NotNil(t, err)
 		require.Equal(t, original, results)
 
-		// TODO: Please update this test in issue https://github.com/grafana/grafana/issues/63686
-		var multierr *multierror.Error
+		var multierr []error
+		multierr = []error{err}
 		require.True(t, errors.As(err, &multierr))
-		require.Equal(t, multierr.Len(), 2)
+		require.Equal(t, len(multierr), 2)
```
Comment (papagian) [inline pkg/services/ngalert/state/cache_test.go:72]: need to replace with:
```
        errs, is := err.(interface{ Unwrap() []error })
		require.True(t, is)
		unwrappedErrors := errs.Unwrap()
		require.Equal(t, len(unwrappedErrors), 2)
```

File: pkg/services/ngalert/state/cache_test.go [hunk 2]
```diff
@@ -67,14 +66,14 @@ func Test_expand(t *testing.T) {
...
 
 		errsStr := []string{
-			multierr.Errors[0].Error(),
-			multierr.Errors[1].Error(),
+			multierr[0].Error(),
+			multierr[1].Error(),
```
Comment (papagian) [inline pkg/services/ngalert/state/cache_test.go:76]: following the above comment
```suggestion
			require.EqualError(t, unwrappedErrors[0], firstErrStr)
		    require.EqualError(t, unwrappedErrors[1], secondErrStr)
```

File: pkg/services/ngalert/state/cache_test.go [hunk 3]
```diff
@@ -103,10 +102,10 @@ func Test_expand(t *testing.T) {
 		require.NotNil(t, err)
 		require.Equal(t, expected, results)
 
-		// TODO: Please update this test in issue https://github.com/grafana/grafana/issues/63686
-		var multierr *multierror.Error
+		var multierr []error
+		multierr = []error{err}
 		require.True(t, errors.As(err, &multierr))
-		require.Equal(t, multierr.Len(), 1)
+		require.Equal(t, len(multierr), 2)
```
Comment (papagian) [inline pkg/services/ngalert/state/cache_test.go:108]: need to replace with:
```
        errs, is := err.(interface{ Unwrap() []error })
		require.True(t, is)
		unwrappedErrors := errs.Unwrap()
		require.Equal(t, len(unwrappedErrors), 1)
```

**Review by papagian — CHANGES_REQUESTED** [review]
I left some suggestions for fixing the tests

**Comment by iSatVeerSingh** [comment]
I don't know why its getting lint errors.

**Review by papagian — CHANGES_REQUESTED** [review]
regarding the lint failures I think it's fine to ignore them for the tests

I cannot reproduce the memcached integration test failure locally and I think it's not related to the changes (must be a flaky test): let's give it a 2nd try

**Review by papagian — APPROVED** [review]
thank you for your contribution!

**Pull request merged**